mos-platform/common/include/fixed_point.h

```diff
@@ -0,0 +1,301 @@
+                                         signed _BitInt((((ISize + FSize) + 7) / 8) * 8),
+                                         unsigned _BitInt((((ISize + FSize) + 7) / 8) * 8)>;
+  StorageType val;
+  static constexpr unsigned long INT_MASK = BitMask<unsigned long>(ISize) << FSize;
+  static constexpr unsigned long FRAC_MASK = BitMask<unsigned long>(FSize);
+
+public:
+
+  // Constructors
+
+  /// Common constructor for creating a fixed point value from an integer value
+  [[clang::always_inline]] constexpr explicit Value(IntType i) {
+    SetI(i); 
+  }
+  
+  /// Copy constructor for a fixed point value with different sizes of Int/Frac
+  template<uint8_t OI, uint8_t OF, bool S>
+  [[clang::always_inline]] constexpr Value(const Value<OI, OF, S>& o) {
+    val = Value<ISize, FSize>(o.AsI(), o.AsF()).Get();
+  }
+  /// Copy constructor for fixed point values with the same size
+  [[clang::always_inline]] constexpr Value(const Value& o) : val(o.val) {}
+  
+  /// Constructor for setting both the integral and fractional part
+  [[clang::always_inline]] constexpr explicit Value(IntType i, FracType f) {
+    SetI(i);
+    SetF(f);
+  }
+  /// Constructor to convert floating point values into fixed point
+  [[clang::always_inline]] constexpr Value(long double f) {
+    Set(f * (1 << FSize));
+  }
+
+  // Direct value accessor and setter methods
+
+  /// Returns just the integral portion
+  [[clang::always_inline]] constexpr IntType AsI() const { return (val & INT_MASK) >> FSize; }
+  /// Returns just the fractional portion
+  [[clang::always_inline]] constexpr FracType AsF() const { return (val & FRAC_MASK); }
+  /// Returns the entire value
+  [[clang::always_inline]] constexpr StorageType Get() const { return val; }
+  /// Update just the integral portion
+  [[clang::always_inline]] constexpr void SetI(IntType v) { 
+    val = (val & ~INT_MASK) | static_cast<StorageType>((v) & (INT_MASK >> FSize)) << FSize;
+  }
+  /// Update just the fractional portion
+  [[clang::always_inline]] constexpr void SetF(FracType v) { 
+    val = (val & ~FRAC_MASK) | (v & FRAC_MASK);
+  }
+  /// Update the entire value
+  [[clang::always_inline]] constexpr void Set(StorageType v) { val = v; }
+
+  // Operator overloads
+  [[clang::always_inline]] constexpr Value& operator=(Value o) {
+    val = o.val;
+    return *this;
+  }
+
+  [[clang::always_inline]] constexpr Value operator +(const Value& o) const {
+    Value<ISize, FSize> n = val;
+    n += o;
+    return n;
+  }
+  [[clang::always_inline]] constexpr Value& operator +=(const Value& o) {
+    val += o.val;
+    return *this;
+  }
+
+  [[clang::always_inline]] constexpr Value operator -(const Value& o) const {
+    Value<ISize, FSize> n = val;
+    n -= o;
+    return n;
+  }
+  [[clang::always_inline]] constexpr Value& operator -=(const Value& o) {
+    val -= o.val;
+    return *this;
+  }
+
+  template<typename T, typename = typename std::enable_if<std::is_arithmetic<T>::value, T>::type>
```

Comment:
The `enable_if` is superfluous here and below; the template will already only expand for types that we can divide the value by. Those are exactly the types that it's meaningful to implement division for.